### PR TITLE
[APEXMALHAR-2220] Move the FunctionOperator to Malhar library

### PR DESCRIPTION
**Pull request by d9liang**
Merge function operators under org.apache.apex.malhar.stream.api.operator and function interface under org.apache.apex.malhar.stream.api.function into org.apache.apex.malhar.lib.function.

### CODE REVIEW

File: library/src/main/java/org/apache/apex/malhar/lib/function/TupleUtil.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.apex.malhar.stream.api.util;
```
Comment (siyuanh) [inline library/src/main/java/org/apache/apex/malhar/lib/function/TupleUtil.java:19]: I think lib.utils is better. It may be used not only in function

Comment (d9liang) [reply library/src/main/java/org/apache/apex/malhar/lib/function/TupleUtil.java:19]: Will change it to lib.utils

**Comment by siyuanh** [comment]
@d9liang Looks good to me, Jenkins failed for some bizarre reason, if travis succeed, could you please squash them and I'll merge it!

**Comment by d9liang** [comment]
The travis mentioned error in malhar-contrib:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-remote-resources-plugin:1.5:process (default) on project malhar-contrib: Error resolving project artifact: Failure to transfer com.splunk:splunk:pom:1.3.0 from https://repository.apache.org/releases/ was cached in the local repository, resolution will not be reattempted until the update interval of sonatype-apache has elapsed or updates are forced. Original error: Could not transfer artifact com.splunk:splunk:pom:1.3.0 from/to sonatype-apache (https://repository.apache.org/releases/): Connect to repository.apache.org:443 [repository.apache.org/207.244.88.143] failed: Connection timed out for project com.splunk:splunk:jar:1.3.0 -> [Help 1]

But there's no problem when I ran it again from my machine:
[INFO] Executed tasks
[INFO]
[INFO] --- build-helper-maven-plugin:1.9.1:attach-artifact (attach-artifacts) @ iteration-demo ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Apache Apex Malhar ................................. SUCCESS [  1.901 s]
[INFO] Apache Apex Malhar Library ......................... SUCCESS [  01:52 h]
[INFO] Apache Apex Malhar Contrib Library ................. SUCCESS [07:03 min]
[INFO] Apache Apex Malhar Demos ........................... SUCCESS [  0.610 s]
[INFO] Apache Apex Malhar MachineData Demo ................ SUCCESS [ 27.917 s]
[INFO] Apache Apex Malhar Pi Demo ......................... SUCCESS [01:01 min]
[INFO] Apache Apex Malhar Twitter Demo .................... SUCCESS [ 59.947 s]
[INFO] Apache Apex Malhar Yahoo! Finance Demo ............. SUCCESS [ 29.232 s]
[INFO] Apache Apex Malhar Fraud Detect Demo ............... SUCCESS [ 18.546 s]
[INFO] Apache Apex Malhar Mobile Demo ..................... SUCCESS [ 16.017 s]
[INFO] Apache Apex Malhar Wordcount Demo .................. SUCCESS [ 27.904 s]
[INFO] Apache Apex Malhar MR Monitoring Demo .............. SUCCESS [ 19.721 s]
[INFO] Apache Apex Malhar MR Operator Demo ................ SUCCESS [ 13.436 s]
[INFO] Apache Apex Malhar Unique Count Demo ............... SUCCESS [ 11.876 s]
[INFO] Apache Apex Malhar R Demo .......................... SUCCESS [ 17.617 s]
[INFO] Apache Apex Malhar EchoServer Demo ................. SUCCESS [ 24.946 s]
[INFO] Apache Apex Malhar Iteration Demo .................. SUCCESS [ 22.169 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:05 h
[INFO] Finished at: 2016-11-03T09:30:08-07:00
[INFO] Final Memory: 443M/1809M
[INFO] ------------------------------------------------------------------------

**Comment by d9liang** [comment]
@siyuanh The build is success from my end. Not sure if I need to try anything else. I've squashed it anyway. Please take a look. Thanks!

**Comment by siyuanh** [comment]
@d9liang I have some compile errors under demo/highlevelapi/WindowedWordCount. Can you take a look?

**Comment by siyuanh** [comment]
@d9liang  looks like the demo code is still using the old package name

**Comment by d9liang** [comment]
@siyuanh I'll look into it.

**Comment by d9liang** [comment]
@siyuanh I've fixed the issue in demos package. I also quashed it again for my PR. The CI check returns good now. Please take a look. Thanks.

**Comment by d9liang** [comment]
@siyuanh bumping up .. let me know if you have any other question

**Comment by d9liang** [comment]
@siyuanh Could you take a look into this? It's been pending for some time. Thanks!

**Comment by d9liang** [comment]
@chandnisingh could you take a look into this PR? It's been pending for quite some time. I didn't hear from Siyuan for two months. It'd be great if you can take over and get it merged soon. Thanks!

**Comment by otterc** [comment]
@d9liang will take a look

**Comment by d9liang** [comment]
@chandnisingh hey did you have a chance to look into it? Thanks.

**Comment by d9liang** [comment]
@ilganeli could you take a look as well? Thanks.

**Comment by ilganeli** [comment]
My primary comment is that there's still dependency on classes in malhar-stream, according to Siyuan we should migrate those dependencies as well unless there's a reason not to.

**Comment by d9liang** [comment]
@ilganeli all the dependent classes that caused cycle dependence problem before have been moved   to the malhar lib. There is no need to move the others.

**Comment by d9liang** [comment]
@ilganeli Please let me know if you or somebody can help merge this soon. Thanks!

**Comment by ilganeli** [comment]
@d9liang LGTM, Don't want to merge unilaterally in case this impacts anyone using these classes. Can someone else please provide a final signoff?

**Comment by d9liang** [comment]
Thanks Ilya. @otterc @siyuanh Ilya has helped look into it. Could you or somebody sign this off please?

**Pull request merged**